jgrapht-core/src/main/java/org/jgrapht/alg/drawing/model/Points.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (C) Copyright 2018-2019, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.drawing.model;
+
+/**
+ * A collection of utilities to assist with point manipulation.
```

Comment:
Is it possible to genericize/generalize this so that it applies beyond Double and Point2D?

Reply:
Not really easy here, as I have a generic parameter array `N[]` as representation, and I am hitting various barriers.